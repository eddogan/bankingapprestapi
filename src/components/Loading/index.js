import React from "react";
// Material-UI imports
import { Box } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";

export default function Loading({ isLoading = true, error = false }) {
  const classes = useStyles();
  if (isLoading && !error) {
    return (
      <div className={classes.loading}>
        <Box component="div" visibility="hidden">
          Loading...
        </Box>
      </div>
    );
  } else if (isLoading && error) {
    return (
      <div className={classes.loading}>
        <Box component="div" className={classes.copy}>
          <small>There was an issue loading this information.</small>
        </Box>
      </div>
    );
  }
  return null;
}

const useStyles = makeStyles(theme => ({
  "@keyframes loading": {
    "0%": { transform: "rotate(0deg)" },
    "100%": { transform: "rotate(360deg)" }
  },
  loading: {
    display: "inline-block",
    position: "relative",
    width: "100%",
    height: "100%",
    "&::after": {
      content: "''",
      position: "absolute",
      left: "48%",
      top: "48%",
      display: "block",
      width: 20,
      height: 20,
      margin: 1,
      borderRadius: "50%",
      border: "2px solid #333",
      borderColor: `${theme.palette.primary.main} transparent ${theme.palette.primary.main}`,
      animation: "loading 1.2s linear infinite",
      animationName: "$loading"
    }
  },
  copy: {
    position: "absolute",
    top: "65%",
    textAlign: "center",
    width: "100%"
  }
}));
