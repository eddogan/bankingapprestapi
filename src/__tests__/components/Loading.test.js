import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Loading from "../../components/Loading";

it("renders correctly", () => {
  const tree = renderer.create(<Loading />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Loading />, div);
  ReactDOM.unmountComponentAtNode(div);
});
