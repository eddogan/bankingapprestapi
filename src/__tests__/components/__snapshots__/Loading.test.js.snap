// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="makeStyles-loading-2"
>
  <div
    className="MuiBox-root MuiBox-root-5"
  >
    Loading...
  </div>
</div>
`;
